crypto/asn1/a_sign.c

```diff
@@ -70,10 +70,15 @@ int ASN1_sign(i2d_of_void *i2d, X509_ALGOR *algor1, X509_ALGOR *algor2,
         }
     }
     inl = i2d(data, NULL);
+    if (inl <= 0) {
```

Comment:
So if `inl` < 0 then the `OPENSSL_clear_free` in the err block is going to do the wrong thing.